File: Easy/3643-flip-square-submatrix-vertically/solution.cpp
class Solution {
public:
    vector<vector<int>> reverseSubmatrix(vector<vector<int>>& grid, int x, int y, int k) {
        for(int j{0}; j < k; ++j)
            for(int i{0}; i< (k/2); ++i) {
                auto tmp = grid[x+i][y+j];
                grid[x+i][y+j] = grid[x+k-i-1][y+j];
                grid[x+k-i-1][y+j] = tmp;
            }
        return grid;
    }
};

File: easy/3643-flip-square-submatrix-vertically/solution.rs
impl Solution {
    pub fn reverse_submatrix(mut grid: Vec<Vec<i32>>, x: i32, y: i32, k: i32) -> Vec<Vec<i32>> {
        (0..k).clone().for_each(|j| (0..k/2).clone().for_each(|i| (grid[(x+i) as usize][(y+j)  as usize], grid[(x+k-i-1) as usize][(y+j) as usize]) = (grid[(x+k-i-1) as usize][(y+j) as usize],grid[(x+i) as usize][(y+j)  as usize])));                
        grid
    }
}